<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>804</width>
    <height>663</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>0</width>
    <height>0</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Demo                                               by WangHongshuo</string>
  </property>
  <property name="autoFillBackground">
   <bool>false</bool>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="minimumSize">
    <size>
     <width>0</width>
     <height>0</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>16777215</width>
     <height>16777215</height>
    </size>
   </property>
   <property name="baseSize">
    <size>
     <width>0</width>
     <height>0</height>
    </size>
   </property>
   <layout class="QGridLayout" name="gridLayout" rowstretch="1,1" columnstretch="1,1,1">
    <item row="0" column="0">
     <layout class="QVBoxLayout" name="verticalLayout" stretch="5,1">
      <item>
       <widget class="ImageWidget" name="show_input_img" native="true">
        <property name="minimumSize">
         <size>
          <width>256</width>
          <height>256</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>500</width>
          <height>500</height>
         </size>
        </property>
        <property name="sizeIncrement">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
        <property name="baseSize">
         <size>
          <width>256</width>
          <height>256</height>
         </size>
        </property>
        <property name="autoFillBackground">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_8">
        <property name="minimumSize">
         <size>
          <width>240</width>
          <height>25</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>500</width>
          <height>25</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>11</pointsize>
          <weight>50</weight>
          <bold>false</bold>
         </font>
        </property>
        <property name="text">
         <string>输入图像</string>
        </property>
        <property name="scaledContents">
         <bool>false</bool>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="0" column="1">
     <layout class="QVBoxLayout" name="verticalLayout_2" stretch="5,1">
      <item>
       <widget class="ImageWidget" name="show_DFT_img" native="true">
        <property name="minimumSize">
         <size>
          <width>256</width>
          <height>256</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>500</width>
          <height>500</height>
         </size>
        </property>
        <property name="sizeIncrement">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
        <property name="baseSize">
         <size>
          <width>256</width>
          <height>256</height>
         </size>
        </property>
        <property name="autoFillBackground">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_9">
        <property name="minimumSize">
         <size>
          <width>240</width>
          <height>25</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>500</width>
          <height>25</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>11</pointsize>
         </font>
        </property>
        <property name="text">
         <string>能量谱</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="0" column="2">
     <layout class="QVBoxLayout" name="verticalLayout_3" stretch="5,1">
      <item>
       <widget class="ImageWidget" name="show_p_DFT_img" native="true">
        <property name="minimumSize">
         <size>
          <width>256</width>
          <height>256</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>500</width>
          <height>500</height>
         </size>
        </property>
        <property name="sizeIncrement">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
        <property name="baseSize">
         <size>
          <width>256</width>
          <height>256</height>
         </size>
        </property>
        <property name="autoFillBackground">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_10">
        <property name="minimumSize">
         <size>
          <width>240</width>
          <height>25</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>500</width>
          <height>25</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>11</pointsize>
         </font>
        </property>
        <property name="text">
         <string>处理后的能量谱</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="1" column="0">
     <layout class="QVBoxLayout" name="verticalLayout_4" stretch="5,1">
      <item>
       <widget class="ImageWidget" name="show_p_img" native="true">
        <property name="minimumSize">
         <size>
          <width>256</width>
          <height>256</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>500</width>
          <height>500</height>
         </size>
        </property>
        <property name="sizeIncrement">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
        <property name="baseSize">
         <size>
          <width>256</width>
          <height>256</height>
         </size>
        </property>
        <property name="autoFillBackground">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_11">
        <property name="minimumSize">
         <size>
          <width>240</width>
          <height>25</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>500</width>
          <height>25</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>11</pointsize>
         </font>
        </property>
        <property name="text">
         <string>处理结果</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="1" column="1">
     <layout class="QVBoxLayout" name="verticalLayout_5" stretch="5,1">
      <item>
       <widget class="ImageWidget" name="show_bw_img" native="true">
        <property name="minimumSize">
         <size>
          <width>256</width>
          <height>256</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>500</width>
          <height>500</height>
         </size>
        </property>
        <property name="sizeIncrement">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
        <property name="baseSize">
         <size>
          <width>256</width>
          <height>256</height>
         </size>
        </property>
        <property name="autoFillBackground">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_12">
        <property name="minimumSize">
         <size>
          <width>240</width>
          <height>25</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>500</width>
          <height>25</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>11</pointsize>
         </font>
        </property>
        <property name="text">
         <string>缺陷增强</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="1" column="2">
     <layout class="QVBoxLayout" name="verticalLayout_6" stretch="10,1">
      <item>
       <widget class="QWidget" name="widget" native="true">
        <property name="minimumSize">
         <size>
          <width>256</width>
          <height>256</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>500</width>
          <height>500</height>
         </size>
        </property>
        <property name="sizeIncrement">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
        <property name="baseSize">
         <size>
          <width>256</width>
          <height>256</height>
         </size>
        </property>
        <layout class="QGridLayout" name="gridLayout_2" rowstretch="1,1,1" columnstretch="1,1">
         <item row="1" column="1">
          <widget class="QPushButton" name="update_date">
           <property name="minimumSize">
            <size>
             <width>116</width>
             <height>23</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>232</width>
             <height>46</height>
            </size>
           </property>
           <property name="text">
            <string>更新</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label">
           <property name="minimumSize">
            <size>
             <width>100</width>
             <height>20</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>100</width>
             <height>20</height>
            </size>
           </property>
           <property name="font">
            <font>
             <pointsize>10</pointsize>
            </font>
           </property>
           <property name="text">
            <string>处理时长：</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QPushButton" name="open_file">
           <property name="minimumSize">
            <size>
             <width>117</width>
             <height>23</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>234</width>
             <height>46</height>
            </size>
           </property>
           <property name="text">
            <string>打开图片</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QLabel" name="p_time">
           <property name="minimumSize">
            <size>
             <width>100</width>
             <height>20</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>100</width>
             <height>20</height>
            </size>
           </property>
           <property name="font">
            <font>
             <pointsize>10</pointsize>
            </font>
           </property>
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
         <item row="0" column="0" colspan="2">
          <widget class="QGroupBox" name="groupBox">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>0</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>240</width>
             <height>155</height>
            </size>
           </property>
           <property name="title">
            <string>参数</string>
           </property>
           <layout class="QFormLayout" name="formLayout">
            <property name="rowWrapPolicy">
             <enum>QFormLayout::DontWrapRows</enum>
            </property>
            <property name="labelAlignment">
             <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
            </property>
            <property name="formAlignment">
             <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
            </property>
            <property name="horizontalSpacing">
             <number>1</number>
            </property>
            <property name="verticalSpacing">
             <number>1</number>
            </property>
            <property name="leftMargin">
             <number>0</number>
            </property>
            <property name="topMargin">
             <number>5</number>
            </property>
            <property name="rightMargin">
             <number>0</number>
            </property>
            <property name="bottomMargin">
             <number>5</number>
            </property>
            <item row="0" column="0">
             <widget class="QLabel" name="label_3">
              <property name="minimumSize">
               <size>
                <width>150</width>
                <height>20</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>300</width>
                <height>40</height>
               </size>
              </property>
              <property name="font">
               <font>
                <pointsize>10</pointsize>
               </font>
              </property>
              <property name="text">
               <string>均值滤波器窗口大小：</string>
              </property>
              <property name="textFormat">
               <enum>Qt::AutoText</enum>
              </property>
              <property name="scaledContents">
               <bool>false</bool>
              </property>
              <property name="alignment">
               <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSpinBox" name="avg_filter_window">
              <property name="minimumSize">
               <size>
                <width>60</width>
                <height>20</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>160</width>
                <height>40</height>
               </size>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_4">
              <property name="minimumSize">
               <size>
                <width>150</width>
                <height>20</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>300</width>
                <height>40</height>
               </size>
              </property>
              <property name="font">
               <font>
                <pointsize>10</pointsize>
               </font>
              </property>
              <property name="text">
               <string>R(邻域半径大小)：</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QSpinBox" name="R">
              <property name="minimumSize">
               <size>
                <width>60</width>
                <height>20</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>160</width>
                <height>40</height>
               </size>
              </property>
              <property name="sizeIncrement">
               <size>
                <width>0</width>
                <height>0</height>
               </size>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_5">
              <property name="minimumSize">
               <size>
                <width>150</width>
                <height>20</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>300</width>
                <height>40</height>
               </size>
              </property>
              <property name="font">
               <font>
                <pointsize>10</pointsize>
               </font>
              </property>
              <property name="text">
               <string>r1(中心保护半径大小)：</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QSpinBox" name="r1">
              <property name="minimumSize">
               <size>
                <width>60</width>
                <height>20</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>160</width>
                <height>40</height>
               </size>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QLabel" name="label_6">
              <property name="minimumSize">
               <size>
                <width>150</width>
                <height>20</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>300</width>
                <height>40</height>
               </size>
              </property>
              <property name="font">
               <font>
                <pointsize>10</pointsize>
               </font>
              </property>
              <property name="text">
               <string>r2(中心去除半径大小)：</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <widget class="QSpinBox" name="r2">
              <property name="minimumSize">
               <size>
                <width>60</width>
                <height>20</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>160</width>
                <height>40</height>
               </size>
              </property>
             </widget>
            </item>
            <item row="4" column="1">
             <widget class="QSpinBox" name="bw_t">
              <property name="minimumSize">
               <size>
                <width>60</width>
                <height>20</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>160</width>
                <height>40</height>
               </size>
              </property>
             </widget>
            </item>
            <item row="5" column="0">
             <widget class="QCheckBox" name="auto_update">
              <property name="minimumSize">
               <size>
                <width>150</width>
                <height>20</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>300</width>
                <height>40</height>
               </size>
              </property>
              <property name="sizeIncrement">
               <size>
                <width>0</width>
                <height>0</height>
               </size>
              </property>
              <property name="text">
               <string>参数变化时自动更新</string>
              </property>
             </widget>
            </item>
            <item row="4" column="0">
             <widget class="QLabel" name="label_7">
              <property name="minimumSize">
               <size>
                <width>150</width>
                <height>20</height>
               </size>
              </property>
              <property name="maximumSize">
               <size>
                <width>300</width>
                <height>40</height>
               </size>
              </property>
              <property name="font">
               <font>
                <pointsize>10</pointsize>
               </font>
              </property>
              <property name="text">
               <string>二值化算法阈值：</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_2">
        <property name="minimumSize">
         <size>
          <width>240</width>
          <height>25</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>500</width>
          <height>25</height>
         </size>
        </property>
        <property name="font">
         <font>
          <pointsize>9</pointsize>
         </font>
        </property>
        <property name="text">
         <string>注：初次处理需分配内存速度较慢，第二次正常。</string>
        </property>
        <property name="scaledContents">
         <bool>false</bool>
        </property>
        <property name="alignment">
         <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>804</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>ImageWidget</class>
   <extends>QWidget</extends>
   <header location="global">ImageWidget-Qt/imagewidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
